src/components/Modal/Modal.tsx

```diff
@@ -0,0 +1,84 @@
+"use client";
+
+import React, { ReactNode } from "react";
+import styled from "styled-components";
+import { Dialog } from "@mui/material";
+import CloseIcon from "@mui/icons-material/Close";
+
+const StyledDialog = styled(Dialog)`
+    & .MuiPaper-root {
+        border: 0;
+        border-radius: 2em;
+        padding: 0;
+        margin: 5%;
+        overflow: hidden;
+        max-width: 90%;
+    }
+`;
+
+const Header = styled.div`
+    display: flex;
+    position: sticky;
+    justify-content: space-between;
+    align-items: center;
+    color: ${(props) => props.theme.secondaryForegroundColor};
+    // TODO VFB-16 Change to using theme with palettes of colours
+    background-color: ${(props) => props.theme.secondaryBackgroundColor}A0;
+    font-size: 1.5em;
+    font-weight: bolder;
+    padding: 0.75em 1em;
+    width: 100%;
+`;
+
+const Content = styled.div`
+    overflow: auto;
+    padding: 1em 1.5em 1.5em;
+    width: 100%;
+`;
+
+const CloseButton = styled.button.attrs({
+    "aria-label": "Close Button",
+})`
+    border: 0;
+    border-radius: 50%;
+
+    // TODO VFB-16 Change to using theme with palettes of colours
+    color: grey;
+
+    display: flex;
+    justify-content: center;
+    aspect-ratio: 1;
+    align-items: center;
+    transition: 0.2s;
+    margin-left: 1em;
+
+    &:hover {
+        // TODO VFB-16 Change to using theme with palettes of colours
+        color: white;
+        background-color: darkgray;
+    }
+`;
+
+interface ModalProps {
+    header: ReactNode;
```

Comment:
I prefer `ReactElement` over `ReactNode`, but try `React.FC` (the best but fiddliest option) first!

Reply:
I think `ReactElement` is best for this situtation as not all things passed to the modal will be a functional component - could be a `string` for instance